url79d4f7c7ed_schanq_Live_Algorithm_tgz-pJ8-algorithms_and_systems_AI_Evolutionary_GeneticAlgorithmJ8/src/expert_system/Feel.java

```diff
@@ -0,0 +1,213 @@
+	
+	STRAIGHTWALK,
+	SWUNGWALK,
+	STRAIGHTSOLO,
+	SWUNGSOLO,
+	LATIN,
+	LASTNOTE,
+	END;
+	
+	public ArrayList <Rhythmic_Block> fitRhythms;
+	
+	// ##########################################################
+	// #					   CONSTRUCTORS						#
+	// ##########################################################
+	private Feel(){
+		fitRhythms = new ArrayList<Rhythmic_Block>();
+		this.initialise();
+	}
+	// ##########################################################
+	// #					   	 METHODS						#
+	// ##########################################################
+	
+	// Returns fitRhythms field
+	public ArrayList<Rhythmic_Block> getFitRhythms() {
+		return fitRhythms;
+	}
+	
+	// Adds the appropriate rhythmic blocks to the fitRhythms arraylist
+	private void initialise(){
+		
+		String name = this.name();
+		
+		if(name == "END"){
+			fitRhythms.add(Rhythmic_Block.END);
+		}
+		else if(name == "LASTNOTE"){
+			fitRhythms.add(Rhythmic_Block.END);
+			fitRhythms.add(Rhythmic_Block.LASTNOTE);
+			
+		}
+		else if(name == "LATIN"){
+			fitRhythms.add(Rhythmic_Block.MINIM1);
+			fitRhythms.add(Rhythmic_Block.CROTCHETS1);
+			fitRhythms.add(Rhythmic_Block.CROTCHETS2);
+			fitRhythms.add(Rhythmic_Block.TRIPLET_QUAVERS_FIRST_OFF);
+			fitRhythms.add(Rhythmic_Block.TRIPLET_QUAVERS_FULL);
+			fitRhythms.add(Rhythmic_Block.TRIPLET_QUAVERS_SECOND_OFF);
+			fitRhythms.add(Rhythmic_Block.TRIPLET_QUAVERS_THIRD_OFF);
+			fitRhythms.add(Rhythmic_Block.LATIN_V1);
+			fitRhythms.add(Rhythmic_Block.LATIN_V2);
+			fitRhythms.add(Rhythmic_Block.LATIN_V3);
+			fitRhythms.add(Rhythmic_Block.LATIN_V4);
+			fitRhythms.add(Rhythmic_Block.LATIN_V5);
+			fitRhythms.add(Rhythmic_Block.LATIN_V6);
+			
+		}
+		else if(name == "SWUNGSOLO"){
+			fitRhythms.add(Rhythmic_Block.CROTCHETS1);
+			fitRhythms.add(Rhythmic_Block.QUAVERS2);
+			fitRhythms.add(Rhythmic_Block.QUAVERS4);
+			fitRhythms.add(Rhythmic_Block.QUAVERS2_FIRST_ON);
+			fitRhythms.add(Rhythmic_Block.QUAVERS2_SECOND_ON);
+			fitRhythms.add(Rhythmic_Block.QUAVERS_SWUNG_T1);
+			fitRhythms.add(Rhythmic_Block.QUAVERS_SWUNG_T2);
+			fitRhythms.add(Rhythmic_Block.QUAVERS_SWUNG_T3);
+			fitRhythms.add(Rhythmic_Block.QUAVERS_SWUNG_T4);
+			fitRhythms.add(Rhythmic_Block.TRIPLET_QUAVERS_FIRST_OFF);
+			fitRhythms.add(Rhythmic_Block.TRIPLET_QUAVERS_FULL);
+			fitRhythms.add(Rhythmic_Block.TRIPLET_QUAVERS_SECOND_OFF);
+			fitRhythms.add(Rhythmic_Block.TRIPLET_QUAVERS_THIRD_OFF);
+			fitRhythms.add(Rhythmic_Block.SEMI_QUAVERS_SWUNG_T1);
+			fitRhythms.add(Rhythmic_Block.SEMI_QUAVERS_SWUNG_T2);
+			fitRhythms.add(Rhythmic_Block.SEMI_QUAVERS_SWUNG_T3);
+			fitRhythms.add(Rhythmic_Block.SEMI_QUAVERS_SWUNG_T4);
+			fitRhythms.add(Rhythmic_Block.TRIPLET_SEMI_QUAVERS_FULL);
+			fitRhythms.add(Rhythmic_Block.TRIPLET_SEMI_QUAVERS_VAR1);
+			fitRhythms.add(Rhythmic_Block.TRIPLET_SEMI_QUAVERS_VAR2);
+			fitRhythms.add(Rhythmic_Block.TRIPLET_SEMI_QUAVERS_VAR3);
+			
+		}
+		else if(name == "STRAIGHTSOLO"){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.